siliconcompiler/schema/schema_cfg.py

```diff
@@ -3774,39 +3770,24 @@ def schema_constraint(cfg):
             (horizontal, vertical) tuple represented in microns.""")
 
     scparam(cfg, ['constraint', 'component', inst, 'rotation'],
-            sctype='float',
+            sctype='enum',
+            enum=['R0', 'R90', 'R180', 'R270',
+                  'MZ', 'MZ_R90', 'MZ_R180', 'MZ_R270',
+                  'MZ_MX', 'MZ_MX_R90', 'MZ_MX_R180', 'MZ_MX_R270',
+                  'MX', 'MX_R90', 'MX_R180', 'MX_R270',
+                  'MY_R90', 'MY_R180', 'MY_R270',
+                  'MZ_MY', 'MZ_MY_R90', 'MZ_MY_R180',  'MZ_MY_R270'],
```

Comment:
it might be nice to have something that documents what each actually is.

Reply:
Fixed